veps/sig-compute/secure-execution.md

```diff
@@ -0,0 +1,59 @@
+# Overview
+Implement support for IBM Z Secure Execution VMs in kubevirt
+
+## Motivation
+To enable customers to deploy secure workloads on their IBM Z machines using kubevirt.
+
+## Goals
+To enable kubevirt to run Secure Execution VMs on s390x
+
+## Non Goals
+Changing how existing VMs run.
+
+## Definition of Users
+- IBM Z users wanting to secure their new VMs using Secure Execution
+- IBM Z users wanting to migrate existing Secure Execution VMs to kubevirt
+
+## User Stories
+(list of user stories this design aims to solve)
+
+## Repos
+https://github.com/kubevirt/kubevirt
+https://github.com/kubevirt/api
+
+# Design
+
+The feature should be behind a Feature Gate at first, similiar to AMD SEV.
+For implementing this feature, the api behaviour would need to be changed to allow for setting an empty `launchSecurity: {}`.
+
+For launching Secure Execution VMs, all devices need to have `iommu=on` set to allow them to be visible to the guest, as in Secure Execution Mode the hypervisor is prevented from accessing the guest memory. Otherwise Secure Execution is initiaded from the guest image during boot, not the hypervisor.
```

Comment:
Would this depend on the architecture, only for IBM Z?

Reply:
Currently AMD SEV is already enabling iommu, so currently it is not. This might change in the future though, as other technologies might have other requirements.

Reply:
Yes, SEV has its own API..
I mean, the IOMMU is being added only if `.launchSecurity.SEV!=nil` right?

Reply:
Yes that is currently the case, but i would change it to `.launchsecurity != nil` since it applies to all current implemented cases.

Reply:
Ok, we can figure this out on the PR.